syntax = "proto2";

message GaniInformation {

  message SpriteDef {
      enum DefaultImageType {
        NOTDEFAULT = 0;
        SPRITES = 1;
        SHIELD = 2;
        HEAD = 3;
        BODY = 4;
        SWORD = 5;
        HORSE = 6;
        PARAM1 = 7;
        ATTR1 = 8;
        PARAM2 = 9;
        ATTR2 = 10;
        PARAM3 = 11;
        ATTR3 = 12;
        }

      optional uint32 index = 1;
      optional string image = 2;
      optional int32 offX = 3;
      optional int32 offY = 4;
      optional int32 width = 5;
      optional int32 height = 6;
      optional string name = 7;
      optional DefaultImageType defaultImage = 8;

      }

  message Sound {
    optional string fileName = 1;
    optional int32 x = 2;
    optional int32 y = 3;
  }

  message AniSprite {
    optional uint32
  }
}